pkg/assembler/backends/keyvalue/search.go

```diff
@@ -18,10 +18,235 @@ package keyvalue
 import (
 	"context"
 	"fmt"
-
 	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+	"strings"
 )
 
 func (c *demoClient) FindSoftware(ctx context.Context, searchText string) ([]model.PackageSourceOrArtifact, error) {
-	return []model.PackageSourceOrArtifact{}, fmt.Errorf("not implemented: FindSoftware")
+	scanner := c.kv.Keys("artifacts")
```

Comment:
can you add tests here: https://github.com/guacsec/guac/tree/main/internal/testing/backend

Reply:
Thanks, added tests